File: service-beacons-go/services/version1/BeaconsCommandableHttpServiceV1.go
package services1

import (
	cref "github.com/pip-services3-go/pip-services3-commons-go/refer"
	cservices "github.com/pip-services3-go/pip-services3-rpc-go/services"
)

type BeaconsCommandableHttpServiceV1 struct {
	*cservices.CommandableHttpService
}

func NewBeaconsCommandableHttpServiceV1() *BeaconsCommandableHttpServiceV1 {
	c := &BeaconsCommandableHttpServiceV1{
		CommandableHttpService: cservices.NewCommandableHttpService("v1/beacons"),
	}
	c.DependencyResolver.Put("controller", cref.NewDescriptor("beacons", "controller", "*", "*", "1.0"))
	return c
}

File: service-beacons-dotnet/src/Service/Services/Version1/BeaconsCommandableHttpServiceV1.cs
using PipServices3.Commons.Refer;
using PipServices3.Rpc.Services;

namespace Nov.MaxSamples.Beacons.Services.Version1
{
    public class BeaconsCommandableHttpServiceV1 : CommandableHttpService
    {
        public BeaconsCommandableHttpServiceV1()
            : base("v1/beacons")
        {
            _dependencyResolver.Put("controller", new Descriptor("beacons", "controller", "default", "*", "1.0"));
        }

        public override void Register()
        {
            if (!_swaggerAuto && _swaggerEnable)
            {
                RegisterOpenApiSpecFromResource("beacons_http_v1.yaml");
            }

            base.Register();
        }
    }
}
